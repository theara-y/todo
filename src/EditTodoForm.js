import { useState } from "react";

const EditTodoForm = ({changeText, currentTodo}) => {
    const INIT = currentTodo;
    const [text, setText] = useState(INIT);

    const handleChange = (e) => {
        setText(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if (text == "") {
            changeText(INIT);
        } else {
            changeText(text);
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={text} onChange={handleChange} placeholder="Edit Todo..."/>
            <button>Save</button>
        </form>
    );
};

export default EditTodoForm;