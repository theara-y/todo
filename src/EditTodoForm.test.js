import { render } from "@testing-library/react";
import EditTodoForm from "./EditTodoForm";

it("renders", () => {
    render(<EditTodoForm/>);
});

it("matches snapshots", () => {
    const { asFragment } = render(<EditTodoForm/>)
    expect(asFragment()).toMatchSnapshot();
})