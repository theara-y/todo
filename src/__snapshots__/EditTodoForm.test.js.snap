// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="Edit Todo..."
      type="text"
      value=""
    />
    <button>
      Save
    </button>
  </form>
</DocumentFragment>
`;
